vcf-dumper/vcf-dumper-ws/src/main/java/uk/ac/ebi/eva/vcfdump/server/rest/HtsgetVcfController.java

```diff
@@ -53,145 +56,182 @@
         }
         if (start != null && referenceName == null) {
-            return ResponseEntity.status(HttpStatus.BAD_REQUEST).body(
-                    Collections.singletonMap("htsget", new HtsGetError("InvalidInput", "Reference name is not specified when start is specified")));
+            return Optional.of(getResponseEntity("InvalidInput", "Reference name is not specified when start is " +
+                    "specified", HttpStatus.BAD_REQUEST));
         }
-        return null;
+        if (referenceName == null) {
+            return Optional.of(getResponseEntity("Unsupported", "'referenceName' is required", HttpStatus.BAD_REQUEST));
+        }
+        return Optional.empty();
+    }
+
+    private ResponseEntity getResponseEntity(String error, String message, HttpStatus httpStatus) {
+        HtsGetError htsGetError = new HtsGetError(error, message);
+        return ResponseEntity.status(httpStatus).body(Collections.singletonMap("htsget", htsGetError));
+    }
+
+    private Optional<ResponseEntity> validateRequest(String referenceName, Long start, Long end,
+                                                     VariantExporterController controller) {
+        if (start == null) {
+            start = controller.getCoordinateOfFirstVariant(referenceName);
+        }
+        if (end == null) {
+            end = controller.getCoordinateOfLastVariant(referenceName);
+        }
+        if (end < start) {
+            // Applies to valid requests such as chromosome 1, start: 1.000.000, end: empty.
+            // If variants exist only in region 200.000 to 800.000, getCoordinateOfLastVariant() will return 800.000.
+            // Given that 800.000 < 1.000.000, no region can be found.
+            return Optional.of(getResponseEntity("NotFound", "The resource requested was not found",
+                                                 HttpStatus.NOT_FOUND));
+        }
+        if (!controller.validateSpecies()) {
+            return Optional.of(getResponseEntity("InvalidInput", "The requested species is not available",
+                                                 HttpStatus.BAD_REQUEST));
+        }
+        if (!controller.validateStudies()) {
+            return Optional.of(getResponseEntity("InvalidInput", "The requested study(ies) is not available",
+                                                 HttpStatus.BAD_REQUEST));
+        }
+        return Optional.empty();
     }
 
     @RequestMapping(value = "/headers", method = RequestMethod.GET, produces = "application/octet-stream")
     public StreamingResponseBody getHtsgetHeaders(
+            @ApiParam(value = "First letter of the genus, followed by the full species name, e.g. ecaballus_20. " +
+                    "Allowed values can be looked up in https://www.ebi.ac.uk/eva/webservices/rest/v1/meta/species/list/" +
+                    " concatenating the fields 'taxonomyCode' and 'assemblyCode' (separated by underscore).",
```

Comment:
Could the swagger descriptions be re-used across the methods (from getHtsgetUrls in this case)?